interface/cuvis.hpp

```diff
@@ -2607,9 +2607,9 @@ namespace cuvis
     return std::string(version);
   }
 
-  inline void General::init(std::string const& settings_path, int global_loglevel)
+  inline void General::init(std::string const& settings_path, int global_loglevel, std::string logfile_name)
   {
-    chk(cuvis_init(settings_path.c_str(), global_loglevel));
+    chk(cuvis_init(settings_path.c_str(), global_loglevel, logfile_name.empty() ? NULL : logfile_name.c_str()));
```

Comment:
```suggestion
    chk(cuvis_init(settings_path.c_str(), global_loglevel, logfile_name.empty() ? nullptr : logfile_name.c_str()));
```
I think the cleaner c++ way would be using nullptr instead of NULL